### ISSUE TITLE
Feature suggestion: Add Gotify as receiver

### ISSUE THREAD

**Issue by etho201** [prometheus/alertmanager#2120]
[Gotify](https://github.com/gotify/server) is a self-hosted server application that supports sending messages via a REST-API. Is there any way to add Gotify as a receiver within Alertmanager?

**Comment by roidelapluie** [comment]
Thanks for suggesting this change. Unfortunately, we are currently not accepting new integrations.

We can only provide the stability and performance we want to provide if we can test integrations in an automated and scalable fashion. For 
this reason, we are suggesting people integrate with the help of our generic interfaces by either integrating natively or putting a rewriting proxy in the middle. We would be more than happy to list this on our integrations page.

Even if existing integrations can not be tested in an automated fashion, we will not remove them for reasons of compatibility. This also means that any additions we take on, or any changes to existing integrations we make or accept, will mean maintaining and testing these until at least the next major version, realistically even beyond that.

Feel free to question this answer on our [developer mailing list](https://groups.google.com/forum/#!forum/prometheus-developers), but be aware it's unlikely that you will get a different answer.

That said, it looks like Gotify supports plugins (https://gotify.net/docs/plugin) and could therefore support out of the box alertmanager wehbooks

**Comment by A1kmm** [comment]
I made a really simple app, available at https://gitlab.com/A1kmm/alertmanager-to-gotify, to send a notification to Gotify for each alert, with the details of the notification controlled by annotations.

**Comment by simonpasquier** [comment]
@A1kmm you can add your project here: https://prometheus.io/docs/operating/integrations/#alertmanager-webhook-receiver

**Comment by DRuggeri** [comment]
I took the same route as @A1kmm, but used go since I'd prefer to avoid docker for my use case:
https://github.com/DRuggeri/alertmanager_gotify_bridge

It works pretty well though isn't completely finished.

I've [opened a PR](https://github.com/prometheus/docs/pull/1717) as suggested by @simonpasquier to add it to the docs.

**Comment by LukasKnuth** [comment]
This issue has been open for years now. Are you accepting new integrations again?

Would you accept a PR if I implemented it directly in alert-manager? Less complexity seems better to me.

**Comment by Tuxdude** [comment]
I build and run my own alertmanager docker images. As part of that, I decided to take the route of building a native integration in alertmanager to gotify rather than running one more service to act as a bridge.

If anyone is interested, you can take a look at this patch - https://github.com/TuxdudeHomeLab/docker-image-alertmanager/blob/master/patches/01-gotify.diff , The patch builds successfully against the latest version of alertmanager, but I didn't get to write any tests. Also, I plan to maintain this patch only as part of my alertmanager docker image. Anyone is welcome to use or reuse it any form that works for them.

If there is interest from the alertmanager team to integrate this, I am happy to put in the effort to upstream it as a proper PR. But I would like to hear the stance first before I put in the extra effort, given the past discussions didn't seem as promising.

**Comment by LukasKnuth** [comment]
I have also built something to get this working: https://github.com/LukasKnuth/gotify-slack-webhook/

It's a plugin for Gotify which will accept Slack Incoming Webhooks (as already supported by AlertManager). It's good enough for my current use-case, have a look if you're interested.

**Comment by najtin** [comment]
I wrote a gotify plugin to directly receive and process webhooks. https://codeberg.org/ocimea/gotify-alertmanager-plugin It aims to keep the code base relatively small. You are welcome to share your thoughts.

**Comment by TheMeier** [comment]
With the plugin from @najtin is this issue sufficiently addressed?

**Comment by TheSp1der** [comment]
> With the plugin from [@najtin](https://github.com/najtin) is this issue sufficiently addressed?

No, the go plugin system is far too fragile for this to be a sufficient option. Maintaining a gotify plugin to provide this support is simply not reliable.

**Comment by disconn3ct** [comment]
Is it because there isn't a big company behind it? The [integrations list](https://prometheus.io/docs/alerting/latest/configuration/#receiver-integration-settings) seems to be entirely for-profit businesses. (Unless I am missing some, only `email`, `rocketchat` and `webhook` out of 18 total listings.)

**Comment by TheSp1der** [comment]
> Unfortunately, we are currently not accepting new integrations.

Yet, other integrations have been proposed and accepted. ([discord](https://github.com/prometheus/alertmanager/commits/main/notify/discord), [mattermost](https://github.com/prometheus/alertmanager/commits/main/notify/mattermost), [incidentio](https://github.com/prometheus/alertmanager/commits/main/notify/incidentio), [mstreams](https://github.com/prometheus/alertmanager/commits/main/notify/msteams), [rocketchat](https://github.com/prometheus/alertmanager/commits/main/notify/rocketchat), and more)

So, the project is clearly accepting new integrations. Let's prepare a PR and get this closed.

**Comment by roidelapluie** [comment]
> Is it because there isn't a big company behind it? The [integrations list](https://prometheus.io/docs/alerting/latest/configuration/#receiver-integration-settings) seems to be entirely for-profit businesses. (Unless I am missing some, only `email`, `rocketchat` and `webhook` out of 18 total listings.)

It is not. We had a multiple-year hiatus into accepting new receivers.

Additionally, open source systems (like gotify) can more easily integrate alertmanager webhooks natively.

**Comment by disconn3ct** [comment]
> We had a multiple-year hiatus into accepting new receivers.

And yet you accepted tons of for-profit receivers during/after that. (A simple "copyright" search in `notify/` shows a [slew in 2019](https://github.com/search?q=repo%3Aprometheus%2Falertmanager%20path%3A%2F%5Enotify%5C%2F%2F%20%22Copyright%202019%22&type=code), when this bug was opened, then [discord](https://github.com/prometheus/alertmanager/commit/550952eb51be17ab9cb5c45b57191efd9a0da8cb) and [sns](https://github.com/prometheus/alertmanager/commit/19e74f9ce76502b06f382013d58815f0c561dafd) in 2021, [webex](https://github.com/prometheus/alertmanager/commit/9e2608ebff5d85a94197416b93cfdd9c662db5e6), [telegram](https://github.com/prometheus/alertmanager/commit/1138a088f60aa8cf4c6ef9c3c3ca23aabfbdd6e0) and [rocketchat](https://github.com/prometheus/alertmanager/commit/8572fe849cd2eceb90b37e8c5ac13aba35ffc2f0) (open-source, but the PR came from grafana.com) in 2022, [jira](https://github.com/prometheus/alertmanager/commit/a509c14b103e97895e7339a1a8cf87d293187b38) and [ms teams](https://github.com/prometheus/alertmanager/commit/d10f21f38c51757fa00f1819fd9db6088de27db7) in 2023, plus teams v2 in 2024, and [incidentio](https://github.com/prometheus/alertmanager/commit/b8fc677c29894e27e8e25c90e3ceb96cc5ccfd3c) last year.

> more easily integrate alertmanager webhooks

What specifically do you think makes the work easier for open source (largely volunteer) developers as opposed to paid developers?

**Comment by TheSp1der** [comment]
Everyone has some good points here, but I'm not trying to start a discussion over where this needs to happen or for-profit vs non-profit receivers. I just want acceptance that if an appropriate PR is opened that it will be considered as other receivers have been in the past. I feel like that's in the best interest of the project, the open-source community, and users. I don't think this is asking a lot.

**Comment by SoloJacobs** [comment]
Hey,

I just merged the webhook example into our docs (all credit goes to @TheMeier ). That doesn't necessarily mean a full receiver is out of the question. But as it stands, the maintainers here are still very strapped for time. So, if the webhook does what it should, I would be very happy. But if something is left open, that is also worth reporting.

Have a nice day everyone!
